src/test/java/games/strategy/util/MatchTest.java

```diff
@@ -109,11 +109,12 @@ public void testGetNMatches_ShouldThrowExceptionWhenMaxIsNegative() {
 
   @Test
   public void testAllMatch() {
```

Comment:
I would recommend having separate test methods for `allMatch()` and `allMatchNotEmpty()`.  Just copy the original one and change the assertion for the empty collection scenario:

```java
@Test
public void testAllMatchNotEmpty() {
  assertFalse("empty collection", Match.allMatchNotEmpty(Arrays.asList(), IS_ZERO_MATCH));
  assertFalse("none match", Match.allMatchNotEmpty(Arrays.asList(-1, 1), IS_ZERO_MATCH));
  assertFalse("some match", Match.allMatchNotEmpty(Arrays.asList(-1, 0, 1), IS_ZERO_MATCH));
  assertTrue("all match (one element)", Match.allMatchNotEmpty(Arrays.asList(0), IS_ZERO_MATCH));
  assertTrue("all match (multiple elements)", Match.allMatchNotEmpty(Arrays.asList(0, 0, 0), IS_ZERO_MATCH));
}

@Test
public void testAllMatch() {
  assertTrue("empty collection", Match.allMatch(Arrays.asList(), IS_ZERO_MATCH));
  assertFalse("none match", Match.allMatch(Arrays.asList(-1, 1), IS_ZERO_MATCH));
  assertFalse("some match", Match.allMatch(Arrays.asList(-1, 0, 1), IS_ZERO_MATCH));
  assertTrue("all match (one element)", Match.allMatch(Arrays.asList(0), IS_ZERO_MATCH));
  assertTrue("all match (multiple elements)", Match.allMatch(Arrays.asList(0, 0, 0), IS_ZERO_MATCH));
}
```

I wouldn't worry about the appearance of duplicate test code here.  Tests are cheap and the documentation for specific methods is worth it in the long term.  Right now someone looking at this test would be hard-pressed to notice the one `allMatch()` in the sea of `allMatchNotEmpty()`s. :smile: